src/ADAL.PCL/Cache/TokenCacheKey.cs

```diff
@@ -120,13 +124,14 @@ public bool Equals(TokenCacheKey other)
         /// </returns>
         public override int GetHashCode()
         {
-            const string Delimiter = ":::";
-            return (this.Authority + Delimiter 
-                + this.Resource.ToLower() + Delimiter
-                + this.ClientId.ToLower() + Delimiter
-                + this.UniqueId + Delimiter
-                + ((this.DisplayableId != null) ? this.DisplayableId.ToLower() : null) + Delimiter
```

Comment:
This is the culprit; `DisplayableId.ToLower()` rather than `DisplayableId.ToLowerInvariant()`.